engine/src/main/scala/cromwell/engine/workflow/workflowstore/WorkflowStoreEngineActor.scala

```diff
@@ -100,6 +107,11 @@ final case class WorkflowStoreEngineActor private(store: WorkflowStore, serviceR
             // given we don't know much about what went wrong here.  `t.getMessage` so the cause propagates to the client.
             sndr ! WorkflowAbortFailureResponse(id, new RuntimeException(s"$message: ${t.getMessage}", t))
         }
+      case AbortAllRunningWorkflowsCommandAndStop =>
+        store.abortAllRunning() map { _ =>
+          log.info(s"Aborting all running workflows.")
+          self ! PoisonPill
```

Comment:
why a `PoisonPill` here?

Reply:
because this is happening in the `map` of a future and I'm worried about calling `context stop self` in here. But I do want the actor to stop once that's done.